### PR TITLE
BrokerService_GetCurrency

### PR DESCRIPTION
**Pull request by denis-kriv**

### CODE REVIEW

File: Services/BrokerService/Controllers/BrokerController.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        public BrokerController([FromServices] ILogger<BrokerController> logger)
+        {
+            this.logger = logger;
+        }
+
+        [Route("getcurrency")]
+        [HttpPost]
+        public Instrument GetCurrency(
+            [FromServices] IGetCurrencyCommand command,
+            [FromBody] int depth,
```
Comment (ifelseelif) [inline Services/BrokerService/Controllers/BrokerController.cs]: Body only one

**Review by ifelseelif — CHANGES_REQUESTED** [review]

File: Services/BrokerService/Controllers/BrokerController.cs (hunk 1 above)
Comment (denis-kriv) [reply Services/BrokerService/Controllers/BrokerController.cs]: fixed

File: Services/BrokerService/Controllers/BrokerController.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using DTO.MarketBrokerObjects;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace BrokerService.Controllers
+{
+    [Route("[controller]")]
+    public class BrokerController : Controller
```
Comment (nmamro) [inline Services/BrokerService/Controllers/BrokerController.cs:14]: If i'm not mistaken, this service is not a rest one

**Review by nmamro — CHANGES_REQUESTED** [review]

File: Services/BrokerService/Commands/GetAllCurrenciesCommand.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using BrokerService.Interfaces;
+using BrokerService.Utils;
+using DTO;
+using DTO.MarketBrokerObjects;
+using System.Collections.Generic;
+
+
+namespace BrokerService.Commands
+{
+    public class GetAllCurrenciesCommand : IGetAllCurrenciesCommand
```
Comment (dTarasova) [inline Services/BrokerService/Commands/GetAllCurrenciesCommand.cs:10]: Can we use ICommandInteface?

File: Services/BrokerService/Commands/GetCurrencyCommand.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+using System.Linq;
+
+
+namespace BrokerService.Commands
+{
+    public class GetCurrencyCommand : IGetCurrencyCommand
+    {
+        public Instrument Execute(BrokerType broker, string token, int depth, string currency)
+        {
+            return BrokerFactory.Create(broker, token, depth).GetInstruments(InstrumentType.Currency).FirstOrDefault(instrument => instrument.Currency == currency);
```
Comment (dTarasova) [inline Services/BrokerService/Commands/GetCurrencyCommand.cs:14]: too long line

File: Services/BrokerService/Controllers/BrokerController.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using BrokerService.Interfaces;
+using BrokerService.Utils;
+using DTO;
+using DTO.MarketBrokerObjects;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
```
Comment (dTarasova) [inline Services/BrokerService/Controllers/BrokerController.cs:9]: What is this comment for?

File: Services/BrokerService/Controllers/BrokerController.cs (hunk 2 above)
Comment (dTarasova) [reply Services/BrokerService/Controllers/BrokerController.cs:14]: +

File: Services/BrokerService/Utils/RequestInfo.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
+﻿using DTO.MarketBrokerObjects;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BrokerService.Utils
```
Comment (dTarasova) [inline Services/BrokerService/Utils/RequestInfo.cs:7]: Why do this stuff in utils? Not un the request?

**Review by dTarasova — COMMENTED** [review]
I don't understand why do we have to achieve currency if we already can achieve all instruments

**Pull request closed without merging**